### PR TITLE
Feat: Custom SDK names

### LINKED ISSUE #616: SDK names to indicate built for the Unity package
When investigating an event captured by the sentry-cocoa SDK, we inspect the event payload's `sdk.name` property. It says: `sentry.cocoa`. This doesn't help us tell apart the build done through submodule in this repository, packaged with the Unity SDK. That information can help us troubleshoot issues and understand adoption of the SDK (of iOS apps in general vs iOS+Unity).

There was a change in the sentry-native (https://github.com/getsentry/sentry-native/pull/677) to change its name (issue: https://github.com/getsentry/sentry-java/issues/1901) when building for Android. But that won't help us understand if that was a Unity Android build (packaged via this repo, UPM packaging).

The goal: Events captured by installing `sentry-unity` should report `sdk.name` as follows:

- [x] C# errors: `sentry.dotnet.unity`
- [x] Android: Java/Kotlin: `sentry.java.android.unity`
- [x] Android: C, C++: `sentry.native.android.unity`
- [x] iOS: Objective-C, Swift, C, C++: `sentry.cocoa.unity`
- [x] macOS: Objective-C, Swift, C, C++: `sentry.cocoa.unity`
- [x] Windows: C, C++: `sentry.native.unity`

### PR DESCRIPTION
**Pull request by vaind**
Implements #616 

- [x] [Android: Java/Kotlin: `sentry.java.android.unity`](https://sentry.io/organizations/sentry-sdks/issues/3245458320/events/3a85e648a4234aad96f34fd7a812bd47/?project=5439417#sdk)
- [x] [Android: C, C++: `sentry.native.android.unity`](https://sentry.io/organizations/sentry-sdks/issues/3089496422/events/b5e3775fd983413dd9e4f7625e247a2c/?project=5439417#sdk)
- [ ] [iOS: Objective-C, Swift, C, C++: `sentry.cocoa.unity`](#)
- [ ] [macOS: Objective-C, Swift, C, C++: `sentry.cocoa.unity`](#)
- [x] [Windows: C, C++: `sentry.native.unity`](https://sentry.io/organizations/sentry-sdks/issues/3110234864/events/b9e20f39f446460ca25f1304dd63f94c/?project=5439417#sdk)
- [x] [Linux: C, C++: `sentry.native.unity`](https://sentry.io/organizations/sentry-sdks/issues/3262191687/events/902febf0c01743bff5b83f45e034eed7/?project=5439417#sdk)

### Depends on 
* https://github.com/getsentry/sentry-java/pull/2016
* https://github.com/getsentry/sentry-cocoa/pull/1793
* https://github.com/getsentry/sentry-cocoa/pull/1816
* https://github.com/getsentry/sentry-java/pull/2035
* https://github.com/getsentry/sentry-cocoa/pull/1853

### TODOs
* [x] update java SDK after a new v6.0.0 beta release
* [ ] update cocoa SDK (v7.17+)
* [ ] update links above after SDK updates

#skip-changelog

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -171,7 +171,7 @@ Expected to exist:
     <Error Condition="!Exists('$(SentryAndroidRoot)')" Text="Couldn't find the Android root at $(SentryAndroidRoot)."></Error>
     <Message Importance="High" Text="Building Sentry Android SDK."></Message>
 
-    <Exec WorkingDirectory="$(SentryAndroidRoot)" Command="./gradlew :sentry-android-core:assembleRelease :sentry-android-ndk:assembleRelease :sentry:jar --no-daemon --stacktrace --warning-mode none"></Exec>
+    <Exec WorkingDirectory="$(SentryAndroidRoot)" Command="./gradlew -PsentryAndroidSdkName=sentry.native.android.unity :sentry-android-core:assembleRelease :sentry-android-ndk:assembleRelease :sentry:jar --no-daemon --stacktrace --warning-mode none"></Exec>
```
Comment (bruno-garcia) [inline Directory.Build.targets:174]: `sentry.native.android.unity` this going to rename only the `sentry-native` events, right?
The Java layer will be `sentry.java.android.unity`?

File: src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs [hunk 2]
```diff
@@ -97,7 +97,7 @@ internal void ModifyManifest(string basePath)
             }
 
             options!.DiagnosticLogger?.LogDebug("Configuring Sentry options on AndroidManifest: {0}", basePath);
-
+            androidManifest.SetSDK("sentry.java.android.unity");
```
Comment (bruno-garcia) [inline src/Sentry.Unity.Editor/Android/AndroidManifestConfiguration.cs:100]: I think this answers my question :)

File: Directory.Build.targets (hunk 1 above)
Comment (vaind) [reply Directory.Build.targets:174]: correct

**Review by bitsandfoxes — APPROVED** [review]
This is awesome!

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**